includes/CAPx/Drupal/Processors/FieldProcessors/FileFieldProcessor.php

```diff
@@ -73,23 +74,17 @@ public function process($data) {
       // Allow altering as this could get messy.
       drupal_alter('capx_pre_fetch_remote_file', $value);
 
-      // The filename never changes when the file is modified but there are timestamps available.
-      // Check the local entity's timestamp against the one from the API in order to determine if the file has changed.
+      // Check the local entity's uri against the one to be created in order to
+      // determine if the file has changed.
       $field_values = $entity->{$this->fieldName}->raw();
 
-      if (isset($field_values['timestamp']) && isset($value['lastModified'])) {
```

Comment:
So when the etag on the profile changes the update hook is triggered
The timestamp check was added to the files to avoid downloading the file unnecessarily
So when the etag changes but the file doesnt the whole file array stays the same?
And you are sure of that?

If that is true, the unique name would match and we could avoid the download and it would solve the imagecache issue. Does the old file get cleaned up if revisions are not enabled?